src/main/java/org/broadinstitute/hellbender/tools/dataflow/MarkDuplicatesDataflow.java

```diff
@@ -71,7 +90,11 @@ protected void setupPipeline(final Pipeline pipeline) {
         final PCollectionView<SAMFileHeader> headerPcolView = pipeline.apply(Create.of(header)).apply(View.<SAMFileHeader>asSingleton());
 
         final PCollection<GATKRead> preads = readsSource.getReadPCollection(intervals);
-        final PCollection<GATKRead> results = preads.apply(new MarkDuplicatesDataflowTransform(headerPcolView));
+
+        final OpticalDuplicateFinder finder = READ_NAME_REGEX != null ?
```

Comment:
does setting the `finder` variable here work? `READ_NAME_REGEX` will be set dynamically by the CLI system and it will not update the `finder` variable